pinery-miso/src/main/java/ca/on/oicr/pinery/lims/miso/MisoClient.java

```diff
@@ -0,0 +1,1031 @@
+    return models.size() == 1 ? models.get(0) : null;
+  }
+
+  @Override
+  public List<Instrument> getInstruments() {
+    return template.query(queryAllInstruments, instrumentMapper);
+  }
+
+  @Override
+  public Instrument getInstrument(Integer instrumentId) {
+    List<Instrument> instruments = template.query(queryInstrumentById, new Object[] { instrumentId }, instrumentMapper);
+    return instruments.size() == 1 ? instruments.get(0) : null;
+  }
+
+  @Override
+  public List<Instrument> getInstrumentModelInstrument(Integer id) {
+    return template.query(queryInstrumentsByModelId, new Object[] { id }, instrumentMapper);
+  }
+
+  @Override
+  public List<Box> getBoxes() {
+    List<MisoBoxPosition> temps = template.query(queryAllBoxes, boxPositionRowMapper);
+    Map<Long, Box> boxes = new HashMap<>();
+    for (MisoBoxPosition temp : temps) {
+      Box box = boxes.get(temp.getId());
+      if (box == null) {
+        box = temp.getBox();
+        boxes.put(box.getId(), box);
+      }
+      box.getPositions().add(temp.getBoxPosition());
+    }
+    return Lists.newArrayList(boxes.values());
+  }
+
+  private static class InstrumentMapper implements RowMapper<Instrument> {
+
+    @Override
+    public Instrument mapRow(ResultSet rs, int rowNum) throws SQLException {
+      Instrument ins = new DefaultInstrument();
+
+      ins.setId(rs.getInt("instrumentId"));
+      ins.setName(rs.getString("name"));
+      ins.setModelId(rs.getInt("platformId"));
+
+      return ins;
+    }
+
+  }
+
+  private static class InstrumentModelRowMapper implements RowMapper<InstrumentModel> {
+
+    @Override
+    public InstrumentModel mapRow(ResultSet rs, int rowNum) throws SQLException {
+      InstrumentModel m = new DefaultInstrumentModel();
+
+      m.setId(rs.getInt("platformId"));
+      m.setName(rs.getString("instrumentModel"));
+
+      return m;
+    }
+
+  }
+
+  private static class OrderRowMapper implements RowMapper<Order> {
+
+    @Override
+    public Order mapRow(ResultSet rs, int rowNum) throws SQLException {
+      Order o = new DefaultOrder();
+
+      o.setId(rs.getInt("orderId"));
+      o.setPlatform(rs.getString("platform"));
+      o.setCreatedById(rs.getInt("createdById"));
+      o.setCreatedDate(rs.getTimestamp("createdDate"));
+      o.setModifiedById(rs.getInt("modifiedById"));
+      o.setModifiedDate(rs.getTimestamp("modifiedDate"));
+
+      return o;
+    }
+
+  }
+
+  private static class UserRowMapper implements RowMapper<User> {
+
+    @Override
+    public User mapRow(ResultSet rs, int rowNum) throws SQLException {
+      User u = new DefaultUser();
+
+      u.setId(rs.getInt("userId"));
+
+      String fullname = rs.getString("fullname");
+      if (fullname.contains(" ")) {
+        u.setFirstname(fullname.substring(0, fullname.lastIndexOf(" ")));
+        u.setLastname(fullname.substring(fullname.lastIndexOf(" ") + 1));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Put single-quotes around ' ' to use the faster "lastIndexOf(char)" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3027)